src/simulation/elements/ESNG.cpp

```diff
@@ -0,0 +1,154 @@
+#include "simulation/ElementCommon.h"
+
+static int update(UPDATE_FUNC_ARGS);
+static int graphics(GRAPHICS_FUNC_ARGS);
+
+void Element::Element_ESNG()
+{
+	Identifier = "DEFAULT_PT_ESNG";
+	Name = "ESNG";
+	Colour = PIXPACK(0xC4F2EB);
+	MenuVisible = 1;
+	MenuSection = SC_NUCLEAR;
+	Enabled = 1;
+  
+  Advection = 0.0f;
+	AirDrag = 0.00f;
+	AirLoss = 1.00f;
+	Loss = 1.00f;
+	Collision = -0.99f;
+	Gravity = 0.0f;
+	Diffusion = 0.00f;
+	Falldown = 0;
+
+	Flammable = 0;
+	Explosive = 0;
+	Meltable = 0;
+	Hardness = 0;
+
+  HeatConduct = 255;
+  DefaultProperties.temp = 0.0f;
+  
+  Description = "Energetic Singularity. Absorbs heat and releases it as energy on touch. VERY DANGEROUS!";
+  
+	Properties = TYPE_ENERGY;
+
+	Update = &update;
+	Graphics = &graphics;
+}
+
+static int update(UPDATE_FUNC_ARGS)
+{
+	int r, rx, ry, cry, crx, spawncount, nb, rx2, ry2, r2;
+  float angle, v;
+  
+  if (sim->parts[i].temp > 100 && !(sim->aheat_enable))
+    {
+      sim->parts[i].temp -= 100;
+      sim->parts[i].life += 1;
+    }
+  
+  for (rx=-3; rx<4; rx++) 
+		for (ry=-3; ry<4; ry++)
+      {
+        r = pmap[y+ry][x+rx];
+        if (!r) continue;
+        if (sim->parts[ID(r)].life > (sim->parts[i].life+1)) {
+          sim->parts[ID(r)].life -= 1;
+          sim->parts[i].life += 1;
+        }
+        if (sim->parts[ID(r)].temp > 100)
+          {
+            sim->parts[ID(r)].temp -= 100;
+            sim->parts[i].life += 1;
+          }
+      }
+  
+  int rmin = (sim->parts[i].vx || sim->parts[i].vy) ? -8 : -1;
+  int rmax = (sim->parts[i].vx || sim->parts[i].vy) ? 9 : 2;
+  
+	for (rx=rmin; rx<rmax; rx++) 
```

Comment:
I'm unwilling to decipher this, clearly a lot more is going on here than what your first comment explains